### ISSUE TITLE
Reproduce of the result in PickScore

### ISSUE THREAD

**Issue by Shredded-Pork** [yifan123/flow_grpo#38]
I used the following setting:

    config.resolution = 512
    config.sample.train_batch_size = 12
    config.sample.num_image_per_prompt = 24
    config.sample.num_batches_per_epoch = 3
    config.sample.test_batch_size = 16 # # This bs is a special design, the test set has a total of 2048, to make gpu_num*bs*n as close as possible to 2048, because when the number of samples cannot be divided evenly by the number of cards, multi-card will fill the last batch to ensure each card has the same number of samples, affecting gradient synchronization.

    config.train.batch_size = config.sample.train_batch_size
    config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch//2
    config.train.num_inner_epochs = 1
    config.train.timestep_fraction = 0.99
    config.train.beta = 0.001    
    config.train.sft=0.0
    config.train.sft_batch_size=3
    config.sample.kl_reward = 0
    config.sample.global_std=True
    config.train.ema=True
    config.num_epochs = 1000
    config.save_freq = 100 # epoch
    config.eval_freq = 25
    config.save_dir = 'logs/pickscore/sd3.5-M'
    config.reward_fn = {
        # "geneval": 1.0,
        "pickscore": 1.0,
        # "unifiedreward": 0.7,
    }

And I use 4 nnodes 32gpus to train the GRPO in PickScore. I obtain the following result:

<img width="410" height="260" alt="Image" src="https://github.com/user-attachments/assets/6e9b6e8c-e681-400d-a6bd-8704d93b4e14" />

which is far low than the result in your paper. The initial evaluation results is 0.7*26=18.2, but in Figure.7, the result is about 21.7.

**Comment by Shredded-Pork** [comment]
And I wonder the experimental setting in Table.2. Why there are two results of PickScore in the first row? And it seems like the result are evaluated on the different prompts, but there is only a baseline of SD3.5-M.

**Comment by yifan123** [comment]
Thanks for your interest in our work!

This week, I’ll adjust my config to ensure that there are good hyperparameters for different GPU counts.

The biggest issue here is that `config.train.gradient_accumulation_steps = 1` and `config.sample.num_batches_per_epoch = 3`; this means that each data collection step results in three updates, which significantly affects performance. Ideally, each data collection should lead to two updates instead.

Figure 7 shows results **without** the KL term — you should refer to **Figure 8(c)** for results **with** KL.

Thank you very much for your second question — we’ll revise the paper to make this clearer. As noted in the caption of **Figure 1**, the *Task Metric* uses the corresponding test set for the trained task, which in this case is the [PickScore test prompts](https://github.com/yifan123/flow_grpo/blob/main/dataset/pickscore/test.txt). The *Image Quality* and *Preference Score* are measured using DrawBench. The intention of **Table 2** is to compare different columns vertically: the *Task Metric* shows the task-specific performance after training, while *Image Quality* and *Preference Score* indicate whether there is a performance drop on general tasks.

For **32 GPUs**, please use the following parameters:

```
config.resolution = 512
config.sample.train_batch_size = 9
config.sample.num_image_per_prompt = 24
config.sample.num_batches_per_epoch = 4
config.sample.test_batch_size = 16
config.train.beta = 0.001
```

Furthermore, as explained in this [issue](https://github.com/yifan123/flow_grpo/issues/22#issuecomment-3022732047), I also recommend modifying [this line](https://github.com/yifan123/flow_grpo/blob/main/flow_grpo/diffusers_patch/sd3_sde_with_logprob.py#L80): change `return prev_sample, log_prob, prev_sample_mean, std_dev_t * torch.sqrt(-1*dt)`
to `return prev_sample, log_prob, prev_sample_mean, std_dev_t`

Then use this parameter set:

```
config.resolution = 512
config.sample.train_batch_size = 9
config.sample.num_image_per_prompt = 24
config.sample.num_batches_per_epoch = 4
config.sample.test_batch_size = 16
config.train.beta = 0.01
```

I’m looking forward to seeing your results with this configuration.

Regarding the initialization checkpoint — you’re right that it makes a big difference. Please confirm that you’re using **sd3.5-m** and not **sd3-m** — in the paper, we used [stable-diffusion-3.5-medium](https://huggingface.co/stabilityai/stable-diffusion-3.5-medium).

**Comment by yifan123** [comment]
Thank you for your reply. I also reran it yesterday, and currently the reward looks like this.

<img width="288" height="433" alt="Image" src="https://github.com/user-attachments/assets/061a5c6c-062d-4d38-b2b5-ce4c61cd059a" />
